GitExtensions/Program.cs

```diff
@@ -167,6 +168,16 @@ private static string GetWorkingDir(string[] args)
                 }
             }
 
+            if (workingDir == "")
+            {
+                var cwd = Environment.CurrentDirectory;
+                var gitDir = GitModule.FindGitWorkingDir(cwd);
+                if (GitModule.IsValidGitWorkingDir(gitDir))
+                {
+                    workingDir = gitDir;
```

Comment:
This also supports command line actions where the construction of a module/commands happens slightly differently. In such cases, `workingDir` would still be an empty string here. If you run this from a repository however, you'd expect that repo to be used. This code tries to find one relative to cwd, and verifies it's a valid repo.

Reply:
this essentially rolls back #4740 
please submit this commit as a separate PR, in it please add a comment why this block of code is needed so in a future we don't repeat #4740 again